### PR TITLE
Brass knuckles: finally working edition (now with rebalances and more fun)

### PR DESCRIPTION
**Pull request by c420-o**
## About The Pull Request
I made this like three times but the other ones were on outdated and fucked up repos. this one works like it's supposed to, it's tested, rebalanced, sprited, cleaned up the code, et cetera. 

## Why It's Good For The Game
there is no longer a continuity error with the "punch shit" poster. now brass knuckles are canon, and they're balanced so you can fuck people up without being too overpowered, if that makes sense. merge if you wanna be a badass and punch shit. 

## Changelog
:cl:
add: Adds the brass knuckles
/:cl:

### CODE REVIEW

**Comment by c420-o** [comment]
I swear this is the last time I'll make a seperate repo for a fucked up master

File: code/datums/martial/brassknuckles.dm [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	var/obj/item/bodypart/affecting = D.get_bodypart(ran_zone(A.zone_selected))
+	var/armor_block = D.run_armor_check(affecting, MELEE)
+	var/picked_hit_type = pick("punch", "smoke", "knuck", "knuckledust")
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(D), 'sound/weapons/punch1.ogg', 25, TRUE, -1)
+	D.visible_message("<span class='danger'>[A] [picked_hit_type]s [D]!</span>", \
+					"<span class='userdanger'>You're [picked_hit_type]ed by [A]!</span>", "<span class='hear'>You hear a sickening sound of flesh hitting flesh!</span>", COMBAT_MESSAGE_RANGE, A)
+	to_chat(A, "<span class='danger'>You [picked_hit_type] [D]!</span>")
+	log_combat(A, D, "punched")
+	D.apply_damage(rand(5,10), BRUTE, affecting, armor_block, wound_bonus = 15)
```
Comment (interestingusernam3) [inline code/datums/martial/brassknuckles.dm:15]: This on average deals less damage than a scalpel while having the same wound_bonus, FYI.

File: code/datums/martial/brassknuckles.dm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		style.teach(user, TRUE)
+
+/obj/item/clothing/gloves/brassknuckles/dropped(mob/user)
+	. = ..()
+	if(user.get_item_by_slot(ITEM_SLOT_GLOVES) == src)
+		style.remove(user)
+
+/obj/item/clothing/gloves/brassknuckles
+	name = "Brass Knuckles"
+	desc = "a contraband item meant to prove weapons are for pussies. Perfect for any bloodthirsty soldier or lunatics wearing tiger masks."
```
Comment (ATH1909) [inline code/datums/martial/brassknuckles.dm:33]:
```suggestion
	desc = "A contraband item meant to prove that weapons are for pussies. Perfect for any bloodthirsty soldier or lunatics wearing tiger masks."
```

File: code/datums/martial/brassknuckles.dm (hunk 1 above)
Comment (c420-o) [reply code/datums/martial/brassknuckles.dm:15]: Yes but the difference is it's blunt wounds so all it does is break bones

File: code/datums/martial/brassknuckles.dm (hunk 2 above)
Comment (c420-o) [reply code/datums/martial/brassknuckles.dm]: Sure why not

**Comment by c420-o** [comment]
Wait I fucked up

**Comment by necromanceranne** [comment]
PEople webedit all the time. It's when you webedit EVERY PR.

**Review by Twaticus — CHANGES_REQUESTED** [review]
chrome_7vMzLeoVw0
im sorry but nothing about this says brass knuckles

**Comment by Rohesie** [comment]
Very much not a fan of this martial art.

Also, won't the user retain it by putting it on the gloves slot and then moving it to the backpack without dropping it?

**Comment by c420-o** [comment]
> chrome_7vMzLeoVw0
> im sorry but nothing about this says brass knuckles

yeah I know. I'll def change the icons



> Very much not a fan of this martial art.
> 
> Also, won't the user retain it by putting it on the gloves slot and then moving it to the backpack without dropping it?

no?

`/obj/item/clothing/gloves/brassknuckles/dropped(mob/user)
	. = ..()
	if(user.get_item_by_slot(ITEM_SLOT_GLOVES) == src)
		style.remove(user)`

**Comment by Rohesie** [comment]
> no?
> 
> `/obj/item/clothing/gloves/brassknuckles/dropped(mob/user) . = ..() if(user.get_item_by_slot(ITEM_SLOT_GLOVES) == src) style.remove(user)`

Is this called when the item goes back from the gloves slot into the hands slot, and then into the backpack?

**Comment by c420-o** [comment]
> > no?
> > `/obj/item/clothing/gloves/brassknuckles/dropped(mob/user) . = ..() if(user.get_item_by_slot(ITEM_SLOT_GLOVES) == src) style.remove(user)`
> 
> Is this called when the item goes back from the gloves slot into the hands slot, and then into the backpack?

it...gets called when the gloves aren't in the slot...

**Comment by c420-o** [comment]
image
new half-assed inhand icon

**Comment by Rohesie** [comment]
> 
> 
> > > no?
> > > `/obj/item/clothing/gloves/brassknuckles/dropped(mob/user) . = ..() if(user.get_item_by_slot(ITEM_SLOT_GLOVES) == src) style.remove(user)`
> > 
> > 
> > Is this called when the item goes back from the gloves slot into the hands slot, and then into the backpack?
> 
> you can't move gloves from the glove slot directly into the backpack

Do please re-read what I wrote.

**Comment by c420-o** [comment]
> > 
> > 
> > > > no?
> > > > `/obj/item/clothing/gloves/brassknuckles/dropped(mob/user) . = ..() if(user.get_item_by_slot(ITEM_SLOT_GLOVES) == src) style.remove(user)`
> > > 
> > > 
> > > Is this called when the item goes back from the gloves slot into the hands slot, and then into the backpack?
> > 
> > you can't move gloves from the glove slot directly into the backpack
> 
> Do please re-read what I wrote.

Alright I see what you're saying. It's copy pasted krav maga code, so it's unlikely but I'll still test it

**Comment by c420-o** [comment]
So after doing some testing I realized that removing the gloves actually don't remove harm_act. this isn't an issue with krav maga gloves for some reason, So I made it that harm_act is removed as well

**Review by c420-o — COMMENTED** [review]
it's all good to go!

**Comment by Rohesie** [comment]
It's not a good solution to check every time you harm, instead we need to guarantee it gets properly added and removed when the conditions are met.

However, I'll be very honest with you. I dislike the idea of expanding martial arts. They're not in a good shape code-wise right now. So unless you intend to do an out-of-scope big refactor for them, I don't think this PR is going far.

**Comment by c420-o** [comment]
> instead we need to guarantee it gets properly added and removed when the conditions are met.

Well style does that, the harm is just a double check thing. Better to be safe than sorry

>I don't think this PR is going far.

:(((

**Comment by Rohesie** [comment]
I'm very sorry. I've been thinking this over and I'd really rather `/datum/martial_art` wasn't expanded.
Ideally we refactor this before.

Do you want to rework your PR to use some other system, or simply close it?
I leave it up to you, I have no strong opinions on the feature itself, the addition of brass knuckles, though I'd prefer new abilities or weapons were more visible whenever not intended to be used stealthily.

**Comment by c420-o** [comment]
> I'm very sorry. I've been thinking this over and I'd really rather `/datum/martial_art` wasn't expanded.
> Ideally we refactor this before.
> 
> Do you want to rework your PR to use some other system, or simply close it?
> I leave it up to you, I have no strong opinions on the feature itself, the addition of brass knuckles, though I'd prefer new abilities or weapons were more visible whenever not intended to be used stealthily.

I'm gonna try to rework it

**Comment by TheBonded** [comment]
damn
I really wanted this to be added

**Pull request closed without merging**